src/rvsdg/to_cfg.rs

```diff
@@ -461,8 +481,18 @@ impl<'a> RvsdgToCfg<'a> {
                     let output_assigned = self
                         .assign_to_vars(&outputs_for_branch.values, shared_vars.as_ref().unwrap());
 
-                    branch_blocks
-                        .push(self.sequence_results(&[outputs_for_branch, output_assigned]));
+                    if first_passthrough == Some(i) {
+                        // Optimization: if this is the passthrough block, run it first before all other branches
+                        before_if = self.sequence_results(&[
+                            before_if,
+                            outputs_for_branch,
+                            output_assigned,
+                        ]);
+                        branch_blocks.push(self.sequence_results(&[]));
```

Comment:
So, to clarify we are still creating a block for the passthrough case, but now `optimize_direct_jumps` is going to clean that block up?

Reply:
yes!